### PR TITLE
fix(1339): add parameters to validator

### PR DESCRIPTION
**Pull request by adong**
## Context

Some users want to create builds based on some input without changing their screwdriver template yaml files. 

## Objective

```
shared:
    image: node:8

jobs:
    main:
        requires: [~pr, ~commit]
        steps:
            - test: echo ok
        parameters:
            auth_strategy: simple
            user:
                value: sd-bot
                description: User running build
```
See screenshot: 
image

This PR added the ability to users to use customized parameters to build as their pipeline needs.

## References

https://github.com/screwdriver-cd/screwdriver/issues/1339
## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by adong** [comment]
Rebased with master and will review with design team.

**Comment by adong** [comment]
Rebased with latest master

File: app/components/pipeline-event-row/component.js [hunk 1]
```diff
@@ -21,13 +23,31 @@ export default Component.extend({
       return eventProperties['event.workflowGraph'] && !eventProperties['event.isSkipped'];
     }
   }),
+
+  numberOfParameters: computed('event.meta.parameters', {
+    get() {
+      const parameters = this.getWithDefault('event.meta.parameters', {});
+
+      return Object.entries(parameters).length;
```
Comment (DekusDenial) [inline app/components/pipeline-event-row/component.js]: ask yourself why do `Object.entries` if you only want the `length`?

File: app/components/pipeline-start/component.js [hunk 2]
```diff
@@ -1,7 +1,28 @@
 import Component from '@ember/component';
 import { computed } from '@ember/object';
 
+const MAX_NUM_OF_PARAMETERS_ALLOWED = 5;
```
Comment (DekusDenial) [inline app/components/pipeline-start/component.js]: is this related to `MAX_NUM_INLINE_PARAMETERS`? if so, tie them up

File: app/components/pipeline-start/styles.scss [hunk 3]
```diff
@@ -16,3 +16,16 @@ button:hover {
...
   color: $sd-primary;
 }
+
+button.start-with-parameters-button {
+  width: 25px;
+  padding-right: 20px;
+}
+
+.start-with-parameters-menu {
+  float: initial;
```
Comment (DekusDenial) [inline app/components/pipeline-start/styles.scss]: if this is not the same as  `none`, please point me to the documentation that explains what this does cuz i rarely see anyone use it, not even browsers.

File: app/components/pipeline-parameterized-build/styles.scss [hunk 4]
```diff
@@ -0,0 +1,49 @@
+button.start-with-parameters-button {
+  width: 25px;
+  padding-right: 20px;
+}
+
+// .start-with-parameters-menu {
```
Comment (DekusDenial) [inline app/components/pipeline-parameterized-build/styles.scss]: clean up if not used

File: app/components/workflow-graph-d3/component.js [hunk 5]
```diff
@@ -95,6 +95,7 @@ export default Component.extend({
     }
   },
   redraw(data) {
+    if (!data) return;
```
Comment (DekusDenial) [inline app/components/workflow-graph-d3/component.js]: is this a legit case where `data` could be falsy/empty even thought the workflow expects something valid? If that's the case, shouldn't there be a error message to the user?

File: tests/integration/components/pipeline-parameterized-build/component-test.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import { module, test } from 'qunit';
```
Comment (DekusDenial) [inline tests/integration/components/pipeline-parameterized-build/component-test.js]: im surprised that there is only one test change given all the above changes

File: app/components/workflow-graph-d3/component.js (hunk 5 above)
Comment (adong) [reply app/components/workflow-graph-d3/component.js]: Agreed. Some redraw keeps happening, and it would be better to handle it in the upstream function invokes `redraw`

File: app/components/pipeline-start/styles.scss (hunk 3 above)
Comment (adong) [reply app/components/pipeline-start/styles.scss]: Yep, it just so happened to be float's `initial` is `none`. Maybe I should be explicit

File: app/components/pipeline-event-row/component.js (hunk 1 above)
Comment (adong) [reply app/components/pipeline-event-row/component.js]: It should just be `Object.keys`

File: app/components/pipeline-parameterized-build/component.js [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+  /**
+   * buildParameters are expected to be an object consists of key value pairs
+   * @type {Object}
+   */
+  buildParameters: {
+    _started_at: '10/10/2019',
+    _started_by: {
+      value: 'adong',
+      description: 'User running build'
```
Comment (jithin1987) [inline app/components/pipeline-parameterized-build/component.js]: test data ?

Comment (jithin1987) [inline app/components/pipeline-parameterized-build/component.js]: test data ?

File: tests/integration/components/pipeline-parameterized-build/component-test.js (hunk 6 above)
Comment (jithin1987) [reply tests/integration/components/pipeline-parameterized-build/component-test.js]: +1 on more tests. We should have more acceptance tests. Though I realize that we don't have any on pipeline creation path. We can take that in a separate PR though unit tests https://github.com/screwdriver-cd/ui/tree/master/tests/unit could get some love on these changes.

File: app/components/pipeline-parameterized-build/component.js (hunk 7 above)
Comment (adong) [reply app/components/pipeline-parameterized-build/component.js]: It was intended to put as `jsdoc` comment section, and now I've removed it.
